### ISSUE TITLE
Provide HOC withRouter to be feature identical with default nextjs method.

### ISSUE THREAD

**Issue by StarpTech** [isaachinman/next-i18next#316]
Hello,
the current approach to access the router inside a component isn't ideal:

```jsx
import { Router } from '~/i18n';
// in the render function
Router.pathname
```
This has several reasons:

- We can't render isomorphic without to conditonally check browser or SSR.
- Differs from `nextjs` [approach](https://github.com/zeit/next.js/#using-a-higher-order-component)
- Not compatible with [hooks](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect) because only the component state can be watched for changes.

The api should look as follow:

```jsx
import { withRouter } from '~/i18n'
withRouter(MyComponent)
```

Thank you!

**Comment by isaachinman** [comment]
Related: #4, https://github.com/isaachinman/next-i18next/issues/1#issuecomment-438341472.

PRs welcome.

**Comment by capellini** [comment]
I've created a [branch](https://github.com/capellini/next-i18next/tree/with-router) (diff:  https://github.com/isaachinman/next-i18next/compare/master...capellini:with-router) that contains an idea of how it _could_ be done...but it's not very elegant.  Basically, it reaches into context and wraps Next's router with our `wrapRouter` helper in `render`.  I'm pretty sure that this will cause unnecessary re-renders and cause performance issues, but I haven't benchmarked anything yet.  At the very least, maybe it will help spark further discussion on how to do this better.

**Comment by martpie** [comment]
I'd be interested in tackling this @isaachinman (mostly the hook, but I guess if you have one, you can have the other).

Is there some guidance/advice you could share before I start diving in?

**Comment by isaachinman** [comment]
@martpie Great! I believe `useRouter` is the hook, and `withRouter` is a HOC for class components. Could be wrong! Haven't looked into it in awhile.

**Comment by martpie** [comment]
Sorry, I meant, what are the fundamental differences between next-i18next `Router` and Next.js's one (which properties are different, etc) 😄

**Comment by isaachinman** [comment]
Ah, I see! Very little should be different. All we do is [hook into certain methods](https://github.com/isaachinman/next-i18next/blob/09be96fad33d37f38997cf89103a4b7d690518a8/src/router/wrap-router.ts#L39) and change the `href` and `as` in some specific cases.

**Comment by nclsndr** [comment]
Hey guys, did you removed the `withRouter` and `useRouter` feats from this package?

**Comment by isaachinman** [comment]
Hi there, I have just released `next-i18next@8.0.0`, which uses the internationalised routing built into NextJs v10+. This new version of `next-i18next` is quite different, thus making this PR no longer relevant. Thanks!
